### PR TITLE
Feat/Komponent AI-459

### PR DESCRIPTION
**Pull request by jackkav**
### Summary
This PR offers a state helper to solve a vuepress docs issue where having multiple stateful component examples on one docs page became cumbersome to manage with vue instances.  Based on this article https://michaelnthiessen.com/renderless-components-5-wild-experiments
#### Changes made:
* created component
* added examples

### PR Checklist
- [x] Does not introduce dependencies
- [x] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [x] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [x] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [x] **Framework style:** abides by the essential rules in Vue's style guide
- [x] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [x] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

**Comment by darrenjennings** [comment]
Can you address my comments in https://github.com/Kong/kongponents/pull/190#discussion_r411571116?

**Comment by travispamaral** [comment]
I noted [here](https://github.com/Kong/kongponents/pull/190#discussion_r411623740) but I wonder if this should be a docs specific component. Im not sure we need this as its own component vs a docs example wrapper. @darrenjennings thoughts? I don't see us using this anywhere other than here so thinking adding it to `/vuepress/components` would be a better use case (like the Swatch component)

File: docs/components/renderless/komponent.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      <div>
+        <label for="apes">What's your favorite great ape?</label>
+        <select id="apes" v-model="data.selected" onchange="s => data.selected = s">
+          <option disabled value="">Please select one</option>
+          <option>gorillas</option>
+          <option>orangutans</option>
+          <option>chimpanzees</option>
+          <option>humans</option>
+        </select>
+        <i v-if="data.selected">{{ data.selected }} are neat!</i>
```
Comment (darrenjennings) [inline docs/components/renderless/komponent.md:66]: til humans are considered great apes

**Comment by darrenjennings** [comment]
> Im not sure we need this as its own component vs a docs example wrapper. @darrenjennings thoughts? I don't see us using this anywhere other than here so thinking adding it to /vuepress/components would be a better use case (like the Swatch component)

I'm not against it. Worst case scenario we never use it outside the docs?

**Pull request merged**